From c2e005019582254480fb29d56cb2b306759590f0 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Mon, 15 Apr 2013 18:07:00 +0200
Subject: [PATCH] (F18) ship a dummy syslog.target

There are buggy unit files which Require syslog.target.
For F18 resurrect a syslog.target unit as a workaround.

https://bugzilla.redhat.com/show_bug.cgi?id=951957
---
 Makefile.am         |  1 +
 units/syslog.target | 20 ++++++++++++++++++++
 2 files changed, 21 insertions(+)
 create mode 100644 units/syslog.target

diff --git a/Makefile.am b/Makefile.am
index c192cb3..c7b5a4a 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -397,6 +397,7 @@ dist_systemunit_DATA = \
 	units/quotaon.service \
 	units/systemd-ask-password-wall.path \
 	units/systemd-ask-password-console.path \
+	units/syslog.target \
 	units/systemd-udevd-control.socket \
 	units/systemd-udevd-kernel.socket \
 	units/system-update.target \
diff --git a/units/syslog.target b/units/syslog.target
new file mode 100644
index 0000000..96eb155
--- /dev/null
+++ b/units/syslog.target
@@ -0,0 +1,20 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+# This unit file exists only for compatibility with broken unit files
+# that express hard requirement dependencies on syslog.target (i.e.
+# invalid usage of a provides-like target).
+
+[Unit]
+Description=Dummy Syslog target for compatibility with broken unit files
+Documentation=man:systemd.special(7)
+Documentation=http://www.freedesktop.org/wiki/Software/systemd/syslog
+
+# Avoid that we conflict with shutdown.target, so that we can stay
+# until the very end and do not cancel shutdown.target if we should
+# happen to be activated very late.
+DefaultDependencies=no
